file-summarizer/summarize.py

```diff
@@ -0,0 +1,364 @@
+#!/usr/bin/env python3
+import asyncio
+import tiktoken
+from typing import List
+import concurrent.futures
+import sys
+import os
+from helper import load_from_gptscript_workspace, save_to_gptscript_workspace
+
+
+MAX_CONTEXT_TOKENS = 128000
+MAX_OUTPUT_TOKENS = 16384
+OVERHEAD_TOKENS = 2000
+MAX_CHUNK_TOKENS = MAX_CONTEXT_TOKENS - MAX_OUTPUT_TOKENS - OVERHEAD_TOKENS
+CHUNK_OVERLAP_TOKENS = 0
+MAX_WORKERS = 4
+# Check for OPENAI_API_KEY
+if "OPENAI_API_KEY" not in os.environ:
+    sys.exit(
+        "ERROR: OPENAI_API_KEY environment variable not found.\n"
+        "Please set it before running the script, e.g.:\n\n"
+        "  export OPENAI_API_KEY='sk-xxxxxxx'\n"
+    )
+
+if "OPENAI_BASE_URL" not in os.environ:
+    sys.exit(
+        "ERROR: OPENAI_BASE_URL environment variable not found.\n"
+        "Please set it before running the script, e.g.:\n\n"
+        "  export OPENAI_BASE_URL='https://api.openai.com/v1'\n"
+    )
+if "OBOT_DEFAULT_LLM_MODEL" not in os.environ:
+    sys.exit(
+        "ERROR: OBOT_DEFAULT_LLM_MODEL environment variable not found.\n"
+        "Please set it before running the script, e.g.:\n\n"
+        "  export OBOT_DEFAULT_LLM_MODEL='gpt-4o'\n"
+    )
```

Comment:
If this environment variable is not set, then we don't want to error. You should pick a default value.

Reply:
set the default model to `gpt-4o`